plan/new.go

```diff
@@ -52,6 +57,27 @@ func buildTestCases(plan *Plan) []TestCase {
 				Plan:      plan,
 				Client:    client,
 				Arguments: testArgs,
+				Skip:      false,
+			}
+
+			for _, filter := range behavior.Filters {
+				// Each filter should be a complete match.
+				matchCnt := 0
+				for _, match := range filter.AxisMatches {
+					if (match.Name == behavior.ClientAxis && match.Value == client) ||
+						testArgs[match.Name] == match.Value {
```

Comment:
Most of this logic could be on the filter so that this just becomes,

    if filter.Matches(testArgs) {
      t.Skip = true
      ..
      continue
    }

You can even have an `AxisMatcher.Matches` function so that filter matcher just becomes,

    for _, m := range matchers {
      if !m.Matches(args) {
        return false
      }
    }
    return true

Also, note that testArgs already contains the ClientAxis and the behavior so you don't have to explicitly check for that.